### DISCUSSION TITLE
Simple etching mask generation macro

### DISCUSSION THREAD

**Discussion by srjmas** [iqm-finland/KQCircuits#23] [Q&A] [answered]
While I understand the design elegancy achieved by the "base metal addition" layer, it requires an additional post processing step at mask preparation. Is there a macro that does just that - generating an etching mask layer from the currently open layout?

**Comment by jheinsoo** [comment, marked as answer]
Hi! I think we need a bit of extra explanation about what do you try to achieve to understand your problem. 

Maybe it is important to be on the the same page about the workflow already supported and see if this actually meets your requirements. In the examples available in KQC, one first designs a single chip. Then, the waver scale optical lithography mask has many chips on it. There are code structures in KQC to define masks and to export a set of files for fabrication. This fileset includes `gds` and `oas` files. In KQC, one can choose 
* layers used for a single export file
* the polarity of the geometry (if filled polygon is dark on the mask or if the non-filled area is etched
* if mirror is needed before export (depends on which side of the physical mask will be facing the wafer)

If your way of working fits this workflow, you do not need to do any manual boolean operations with the layers, KQC does it for you. If you do want to make any manual boolean operations, such as "union" or "inverse", KLayout supports them with simple macros or [GUI features](https://www.klayout.de/doc-qt5/manual/layer_boolean.html).

**Reply by srjmas** [reply]
Actually the KLayout Layer Boolean Operations you suggested work well, thank you

**Comment by srjmas** [comment]
Thank you for your detailed answer. Polarity and mirroring are always
fairly easily handled at the lithography tool. I guess what we need for our
CAD based workflow is just a simple macro creating a flat etch layer for
all the active layout by combining and substracting all the required design
layers. The flat etch layer should be located in a new separate cell under
the top cell.

Can you help me making such a macro or point me to the relevant Klayout
documentation? What are the layers contributing to the etch layer?
